tests/cbmc/stubs/madvise.c

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use
+ * this file except in compliance with the License. A copy of the License is
+ * located at
+ *
+ *     http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cbmc_proof/nondet.h>
+
+int madvise(void *addr, size_t length, int advice)
+{
+    return nondet_int();
+}
```

Comment:
Is there any property we could assert for `addr`, `length, or `advice`? We should set `errno` nondeterministically, right? Take a look at this doc https://man7.org/linux/man-pages/man2/madvise.2.html

Reply:
If I remember right, we also did this in C-ESDK. By returning non-deterministically we cover more code-paths and emulate strange madvise behavior on certain architectures. It also adds little to no overhead.